File: dungeons-1.21.11/src/main/java/fishmod/features/BridgeBot.java
package fishmod.features;

import fishmod.utils.Misc;
import fishmod.utils.config.values.FishSettings;
import net.fabricmc.fabric.api.client.message.v1.ClientReceiveMessageEvents;
import net.minecraft.text.Text;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class BridgeBot {

    // Matches guild chat after stripping colors:
    // "Guild > [optional rank] BotName: [optional rank] PlayerName: message"
    // Also handles Discord bridge style "Guild > BotName: Player > message"
    private static Pattern pattern = buildPattern();

    private static Pattern buildPattern() {
        if (FishSettings.bridgeBotName.isBlank()) return Pattern.compile("(?!)"); // never matches
        String bot = Pattern.quote(FishSettings.bridgeBotName.trim());
        // "Guild > [rank] BotName [guild_rank]: PlayerName » message"
        // rank before bot and guild_rank after bot are both optional
        // separator between player and message can be »  :  >  |  etc.
        return Pattern.compile(
            "^Guild > (?:\\[\\S+\\] )?" + bot + "(?:\\s+\\[[^\\]]+\\])?: (\\S+)[^\\w]+(.+)$"
        );
    }

    public static void rebuildPattern() {
        pattern = buildPattern();
    }

    public static void init() {
        ClientReceiveMessageEvents.ALLOW_GAME.register((message, overlay) -> {
            if (overlay || !FishSettings.bridgeBotEnabled || FishSettings.bridgeBotName.isBlank()) return true;

            String plain = message.getString().replaceAll("§.", "").trim();
            Matcher m = pattern.matcher(plain);
            if (!m.matches()) return true;

            String player = m.group(1);
            String text   = m.group(2);
            Misc.addChatMessage(Text.literal("§2Guild > §r§a[Bridge] §r" + player + "§r: " + text));
            return false; // suppress original bot message
        });
    }
}

File: dungeons-26.1.2/src/main/kotlin/fishmod/features/BridgeBot.kt
package fishmod.features

import fishmod.utils.Misc
import fishmod.utils.config.values.FishSettings
import net.fabricmc.fabric.api.client.message.v1.ClientReceiveMessageEvents
import net.minecraft.network.chat.Component
import java.util.regex.Pattern

object BridgeBot {

    private val COLOR = fishmod.utils.Constants.STRIP_COLOR_REGEX

    @Volatile private var pattern: Pattern = buildPattern()

    private fun buildPattern(): Pattern {
        val name = FishSettings.bridgeBotName.trim()
        if (name.isBlank()) return Pattern.compile("(?!)")
        val bot = Pattern.quote(name)
        return Pattern.compile("^Guild > (?:\\[\\S+] )?$bot(?:\\s+\\[[^\\]]+])?[:\\s]+([^\\s:»]+)\\s*[»:]\\s*(.+)$")
    }

    @JvmStatic
    fun rebuildPattern() { pattern = buildPattern() }

    @JvmStatic
    fun init() {
        rebuildPattern()
        ClientReceiveMessageEvents.ALLOW_GAME.register { message, overlay ->
            if (overlay || !FishSettings.bridgeBotEnabled || FishSettings.bridgeBotName.isBlank()) return@register true
            val plain = message.string.replace(COLOR, "").trim()
            val m = pattern.matcher(plain)
            if (!m.matches()) return@register true
            Misc.addChatMessage(Component.literal("§2Bridge > §r${m.group(1)}§r: ${m.group(2)}"))
            false
        }
    }
}
